src/rgw/rgw_op.cc

```diff
@@ -1250,8 +1250,21 @@ int RGWGetObj::handle_user_manifest(const char *prefix)
     return 0;
   }
 
+  r = RGWRados::Object::Read::range_to_ofs(s->obj_size, ofs, end);
+  if (r < 0) {
+    return r;
+  }
+
+  r = iterate_user_manifest_parts(s->cct, store, ofs, end,
+        pbucket_info, obj_prefix, bucket_acl, *bucket_policy,
+        &total_len, nullptr, nullptr,
+        nullptr, nullptr);
```

Comment:
this extra call to `iterate_user_manifest_parts()` is needed to support negative offsets?